src/org.hdfgroup.object/hdf/object/Attribute.java

```diff
@@ -68,26 +68,12 @@
  * @version 2.0 4/2/2018
  * @author Peter X. Cao, Jordan T. Henderson
  */
-public class Attribute extends Dataset implements DataFormat, CompoundDataFormat {
```

Comment:
While I think this is definitely the right direction, I think this is backwards from what we should have. Intuitively, it doesn't make sense to me that an Attribute would extend an AttributeDataset. Rather, an AttributeDataset should be an extension of the Attribute base class, where the AttributeDataset class implements some sort of "Dataset" interface that contains all of the methods needed to read/write. Those should all probably just be moved into the DataFormat interface and AttributeDataset can continue to implement that. Consider that an AttributeDataset is probably something that the TableViews should be working with, whereas non-Tableview classes (like the TreeView)  should only be working with the Attribute base class, without knowing about the I/O related methods.

Reply:
So I still needed the old Attribute class until I had the refactoring done. That is why the naming was kind of reversed. Yes, the naming needs lots of work along with code locations. Given that I am still fixing some test issues, it would be nice to design the process to fix this issue afterwards.

Reply:
Sure, I think this is just a case of not really needing to separate this into a class and subclass though. Since the original Attribute class already extended the Dataset class, we theoretically should have had everything needed to correctly implement the reading/writing of attributes. I think separating this into Attribute and AttributeDataset (or whatever the final names are), is just going to make things more confusing in the long run. For example,

`// Create a new attribute
AttributeDataset dataRange = new Attribute(name, type, dims);`

really just looks like it's going to have people wondering whether they should be instantiating a new AttributeDataset or Attribute depending on their use case.

I think there's certainly a case here for Attribute and CompoundAttribute classes, following the same pattern as regular Datasets, but otherwise it seems like everything changed here can still just all go into the Attribute class.

Reply:
I see what you mean, I will look why I didn't combine them. Just too involved in the process to notice, likely.

Reply:
So AttributeDataset is between Attribute and Dataset, functions needed for an Attribute with data handling like datasets. Attribute major difference is it treats attribute data like strings. And non-hdf5 data needs that, so maybe Atrribute should be renamed AttributeStrings?

Reply:
Of course Attribute needs the common functions in AttributeDataset, and handles both Scalar and Compound types, unlike H5ScalarAttr H5CompoundAttr

Reply:
Updated Attribute class structure.
FitsAttribute isn't required as there are no attributes for fits, but there is a getMetadata function that needs corrected first. So I created a class, FitsAttribute, to satisfy the code in that function.